### ISSUE TITLE
bug: Local Notifications don't work when app is inactive

### ISSUE THREAD

**Issue by pitis** [ionic-team/capacitor#2864]
# Bug Report

## Capacitor Version

`npx cap doctor` output:

  @capacitor/cli: 2.0.2    
  @capacitor/core: 2.0.2   
  @capacitor/android: 2.0.2
  @capacitor/electron: 2.0.2
  @capacitor/ios: 2.0.2

Installed Dependencies:

  @capacitor/electron not installed
  @capacitor/cli 2.0.2
  @capacitor/android 2.0.2
  @capacitor/ios 2.0.2
  @capacitor/core 2.0.2

## Affected Platform(s)

- [ X ] Android
- [ X ] iOS
- [ ] Electron
- [ ] Web

## Current Behavior

In the iOS version of the app, I use 'import { LocalNotifications } from '@ionic-native/local-notifications' in React to receive notifications. Problem is, I only receive the notifications when I'm in app, when the app is inactive, no notifications appear, either on lockscreen or at the top of the display.

## Expected Behavior
I am able to receive local notifications anytime, even if the app is inactive, in the background, or dead.

## Reproduction Steps

After I import LocalNotifications from @ionic-native/local-notifications

    let localNotifs = LocalNotifications

    localNotifs.schedule({
          id: Math.floor(Math.random() * 1000 + 1),
          title: `New message from ${data.message.chatData.sender.name}`,
          text: data.message.message,
          icon: data.message.chatData.sender.profileImg,
          foreground: true,
        })

## Other Technical Details

I tried to use cordova plugin with capacitor, but also it doesn't work. I can't get those notifications to work on iOS.

`npm --version` output: 6.14.4

`node --version` output: v13.8.0

`pod --version` output (iOS issues only): 1.9.1

## Other Information

I'm using socketIO to receive some messages, and I receive them, but only if the app is active. When the app is in foreground, I don't receive them. On Android it works for a while, but after 10 mins I don't receive messages anymore, although the app wasn't even killed.

**Comment by jcesarmobile** [comment]
the problem is the webview doesn't execute js in background, not that the notifications don't work, if you schedule them when in foreground they will appear at the time they were scheduled.

**Comment by pitis** [comment]
> the problem is the webview doesn't execute js in background, not that the notifications don't work, if you schedule them when in foreground they will appear at the time they were scheduled.

but on Android it works for a while. So then we could not have normal notifications like on WhatsApp with Ionic?

**Comment by jcesarmobile** [comment]
whatsapp uses push notifications, not local notifications

**Comment by pitis** [comment]
> whatsapp uses push notifications, not local notifications

and do those work with Ionic? If app is closed, will it receive notification about new message sent from socketIO?

**Comment by fhrbek** [comment]
@pitis this might be unrelated, but I came here when solving a similar (but yet different) issue. I couldn't see any notifications after an iOS application was suspended.
The reason for this was that on suspend, I generated too many notifications in advance and iOS has a limit of 64 notifications, so the earliest notifications were silently cancelled and never appeared.
A simple fix is not to plan more than 64 notifications at a time. Sadly, it's not very well documented anywhere, I found this information at stackoverflow.
So, although it's probably a different issue than yours, this comment may help others landing here if they look for an issue like mine.
